### ISSUE TITLE
I think it's a cool idea, and I appreciate you sharing it, but you should have re-measured the PSNR metrics of other algorithms yourself, because it looks strange, particularly the EDSR-related metrics, and your metrics aren't very comparable to CARN or IMDN.

### ISSUE THREAD

**Issue by Jchenlong** [supercaoO/MSWSR#1]

**Comment by Jchenlong** [comment]
I think it's a cool idea, and I appreciate you sharing it, but you should have re-measured the PSNR metrics of other algorithms yourself, because it looks strange, particularly the EDSR-related metrics, and your metrics aren't very comparable to CARN or IMDN.

**Comment by supercaoO** [comment]
@Jchenlong Thanks.

Referring to our paper, "Considering that single-scale SR methods require multiple networks to perform multi-scale SR, it fails to meet the requirements of multi-scale SR via a single network. For fair comparisons, we make these compared methods to support (2, 4, 8)× SR tasks via a single network. More specifically, we first compute the numbers of parameters and 2× SR results based on their official 2× SR networks. Then we cascade the single 2× SR network n times to obtain the 2n× SR network for computing FLOPs, inference time, and PSNR/SSIM/LPIPS results. Compared with using multiple networks to support (2, 4, 8)× SR, the above procedure can reduce their total number of network parameters by using a single 2× network."

Therefore, in our paper, the PSNR metrics achieved by single-scale SR methods in the aforementioned setting are not the same as those in their original papers.

For fair comparisons, maybe we should compare MSWSR with the CARN and IMDN in the aforementioned setting rather than the CARN and IMDN in their papers.

**Comment by Jchenlong** [comment]
> @Jchenlong谢谢。
> 
> 关于我们的论文，“考虑到单尺度SR方法需要多个网络来执行多尺度SR，它无法通过单个网络满足多尺度SR的要求。为了公平地进行比较，我们将这些比较的方法支持通过单个网络执行（2、4、8）×SR任务，具体而言，我们首先根据其官方2×SR网络计算参数数量和2×SR结果，然后将单个2×SR网络级联获得用于计算FLOP，推理时间和PSNR / SSIM / LPIPS结果的2n×SR网络，与使用多个网络支持（2、4、8）×SR相比，上述过程可以减少它们的网络参数总数通过使用单个2x网络。”
> 
> 因此，在我们的论文中，在上述设置下通过单尺度SR方法获得的PSNR指标与其原始论文中的指标不同。
> 
> 为了进行公平的比较，也许我们应该将MSWSR与上述设置中的CARN和IMDN进行比较，而不是将其论文中的CARN和IMDN进行比较。

Thank you for your reply. Following your way, I went and tried CARN and IMDN. In the Set14 dataset, the PSNR for x4 was 28.46 and 28.49, respectively. You used additional training data, downsampling quadruples and octuplets of low-resolution images. I think it's unfair to them that you only used a pre-trained model for x2. They aren't progressively upsampled per se. You have an advantage in that you can generate multi-scale images, but they don't, and LapSRN and DRN don't use this comparison.

**Comment by Jchenlong** [comment]
> @Jchenlong Thanks.
> 
> Referring to our paper, "Considering that single-scale SR methods require multiple networks to perform multi-scale SR, it fails to meet the requirements of multi-scale SR via a single network. For fair comparisons, we make these compared methods to support (2, 4, 8)× SR tasks via a single network. More specifically, we first compute the numbers of parameters and 2× SR results based on their official 2× SR networks. Then we cascade the single 2× SR network n times to obtain the 2n× SR network for computing FLOPs, inference time, and PSNR/SSIM/LPIPS results. Compared with using multiple networks to support (2, 4, 8)× SR, the above procedure can reduce their total number of network parameters by using a single 2× network."
> 
> Therefore, in our paper, the PSNR metrics achieved by single-scale SR methods in the aforementioned setting are not the same as those in their original papers.
> 
> For fair comparisons, maybe we should compare MSWSR with the CARN and IMDN in the aforementioned setting rather than the CARN and IMDN in their papers.

Thank you for your reply. Following your way, I went and tried CARN and IMDN. In the Set14 dataset, the PSNR for x4 was 28.46 and 28.49, respectively. You used additional training data, downsampling quadruples and octuplets of low-resolution images. I think it's unfair to them that you only used a pre-trained model for x2. They aren't progressively upsampled per se. You have an advantage in that you can generate multi-scale images, but they don't, and LapSRN and DRN don't use this comparison.

**Comment by supercaoO** [comment]
@Jchenlong Thanks for the discussion!

Most of the single-scale SR methods like VDSR and EDSR aim to train one model for one specific scale, while our MSWSR tries to train one model for several scales simultaneously. The original goals of these single-scale SR methods and our MSWSR are different, hence, the comparison between them is hard to be absolutely fair. Since our paper focuses on "multi-scale SR via a single network", that's why the comparison with single-scale SR methods is conducted under the experimental setting of "multi-scale SR via a single network".

However, the aforementioned setting is not used in the comparison with multi-scale SR methods such as LapSRN. The PSNR of LapSRN shown in our paper is totally the same as the PSNR in the original paper of LapSRN.

MSWSR is our exploration of multi-scale SR vias a single network. In the future, we will focus on arbitrary-scale SR via a single network.

**Comment by Jchenlong** [comment]
Absolute fairness is difficult to come by. Thanks for your response; there are a lot of benefits, and the cascade is a very cute idea; the produced results are surprising to me. I'm excited to see what you come up with next!
